apps/process_tpstream.cxx

```diff
@@ -159,21 +167,28 @@ int main(int argc, char const *argv[])
 
   auto channel_map = dunedaq::detchannelmaps::make_map(channel_map_name);
   // Finally create a TA maker
-  // Waiting for A.Oranday's factory!
-  triggeralgs::TriggerActivityMakerHorizontalMuon hmta;
+  auto ta_factory = triggeralgs::TriggerActivityFactory::get_instance();
+  auto ta_maker = ta_factory->build_maker(plugin_name);
   // We should config the algo, really
-  const nlohmann::json config = nlohmann::json::parse(R"(
-    {
-      "trigger_on_adc": false,
-      "trigger_on_n_channels": false,
-      "trigger_on_tot": false,
-      "trigger_on_adjacency": true,
-      "adjacency_threshold": 100,
-      "adj_tolerance": 3
-    }
-  )");
+  nlohmann::json config;
+  if (config_name.length()) {
+    std::ifstream config_stream(config_name);
+    config = nlohmann::json::parse(config_stream);
+  } else {
+    config = nlohmann::json::parse(R"(
+      {
+        "trigger_on_adc": false,
+        "trigger_on_n_channels": false,
+        "trigger_on_tot": false,
+        "trigger_on_adjacency": true,
+        "adjacency_threshold": 100,
+        "adj_tolerance": 3,
+        "prescale": 1
+      }
+    )");
+  }
```

Comment:
I would change this to accept configs as generated by daqconf, including plugin name. That way we would be able to literally provide the config that we ran on e.g. coldbox with, without changing anything. 
So that would change to:
```
{
    "trigger": {
        "trigger_activity_plugin": [
            "TriggerActivityMakerHorizontalMuonPlugin"
        ],
        "trigger_activity_config": [
            {
                "trigger_on_adc": false,
                "trigger_on_n_channels": false,
                "trigger_on_tot": false,
                "trigger_on_adjacency": true,
                "adjacency_threshold": 100,
                "adj_tolerance": 3,
                "prescale": 1,
                "window_length": 4000
            }
        ]
    }
}
``` 
,with ability to pull algorithm name from the config, and maybe some checks so it does't allow for multiple TAMakers/algorithms (or change it so the script CAN work on multiple algorithms), maybe even include some default config in a separate file.

That would be maybe for the next PR though, not this one -- this one is approved :)

Reply:
Yes. This PR only makes changes to `trgtools`. This is only referencing the changes in other repositories that make this PR possible.